filters/scanner.go

```diff
@@ -138,6 +138,10 @@ chomp:
 		return pos, tokenValue, s.input[pos:s.ppos]
 	case isFieldRune(ch):
 		s.scanField()
+		if ch == '-' || isDigitRune(ch) { // illegal token if it starts with '-' or any digit
+			s.error("illegal token field")
```

Comment:
@dmcgowan / @estesp - The implementation was less strict than the syntax here. Do we want to make this change in 1.x? 

https://github.com/containerd/containerd/blob/b2a01eedf4aa6f7a80aaa47a7096b180ee6c36e7/filters/parser.go#L39-L47

If we follow the syntax and update the implementation;
- It would be better to explain why this is illegal in the error itself.
- It should handle `_` as well.